pallene/to_ir.lua

```diff
@@ -145,9 +145,12 @@ function ToIR:resolve_variable(decl)
     return var
 end
 
-function ToIR:register_lambda(exp, name)
+function ToIR:register_lambda(exp, name, is_hof)
```

Comment:
If get rid of the `module.lambdas`, can we also get rid of this `is_hof` argument?